cmd/juju/application/storage.go

```diff
@@ -0,0 +1,333 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/names/v5"
+	"github.com/juju/utils/v3"
+
+	"github.com/juju/juju/api/client/application"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/juju/config"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+	"github.com/juju/juju/storage"
+)
+
+const (
+	appStorageConfigSummary = "Displays or sets storage directives for an application."
+
+	appStorageConfigDoc = `
+To view all storage directives for the given application:
+
+    juju application-storage <application>
+
+	By default, the config will be printed in a tabular format. You can instead
+print it in ` + "`json`" + ` or ` + "`yaml`" + ` format using the ` + "`--format`" + ` flag:
+
+   	juju application-storage <application> --format json
+    juju application-storage <application> --format yaml
+
+To view the directive of a single storage name:
+
+    juju application-storage <application> <storage-name>
+
+To set storage directives on an application:
+
+    juju application-storage <application> name1=size,pool,count name2=size,pool,count
+
+Config values can be imported from a yaml file using the ` + "`--file`" + ` flag:
+
+    juju application-storage <application> --file=path/to/config.yaml
+
+This allows you to, e.g., save an application's storage directives to a file:
+
+    juju application-storage <application> --format=yaml > config.yaml
+
+and then import the config later. 
+`
+	appStorageConfigExamples = `
+Print the storage directives for all storage names of the postgresql application:
+
+    juju application-storage postgresql
+
+Print the storage directives for the storage name 'pgdata' of the postgresql application:
+
+    juju application-storage postgresql pgdata
+
+Set the size to 100GiB, pool name to "rootfs", and count to 1 for the mysql application's 'database' storage:
+
+    juju application-storage mysql database=100G,rootfs,1
+
+If no size is provided, Juju uses the minimum size required by the charm. If the charm does not specify a minimum, the default is 1 GiB. 
+This value is then applied when updating the application’s storage.
+
+    juju application-storage mysql database=,rootfs,1
+
+If no pool is provided, Juju selects the default storage pool from the model.
```

Comment:
This is true, but if the model has no pool defined, then a provider specific default is used. But only if size is specified. If size is not specified, the I think we default to rootfs (for filesystem) or loop (for block).
It's complicated. I wonder if we need this level of detail here. Let's be consistent with how we document `--storage` for deploy and refresh commands, since it's the same process for all.

Reply:
im concerned because i was trying it out myself the first time and i expected, without including pool value, there should be no updates to the pool. However this is not true, it actually updates the pool with the default value. I thought of adding a generic caveat here for the purpose of warning users that a default value will still be passed for the update, what do you think?